collector/internal/lifecycle/manager.go

```diff
@@ -139,10 +146,21 @@ func (lm *manager) processEvents(ctx context.Context) error {
 				return err
 			}
 
-			err = lm.listener.Wait(ctx, res.RequestID)
-			if err != nil {
-				lm.logger.Error("problem waiting for platform.runtimeDone event", zap.Error(err), zap.String("requestID", res.RequestID))
-			}
+			lm.reqID = res.RequestID
+			lm.wg.Wait()
```

Comment:
This doesn't seem safe, given that this function also calls `defer lm.wg.Done()`.  I'd expect this line to block forever.  This should probably be a separate `WaitGroup` that is incremented and waited on here and decremented in the event handler.

Reply:
You're right, I've updated the code to use a separate waitgroup for HandleEvent, as well as a lock for reading/writing the `reqID`. I've also added unit tests around the execution of processEvents and HandleEvent. PTAL